analysis/token_filters/camelcase_filter/parser.go

```diff
@@ -0,0 +1,93 @@
+package camelcase_filter
+
+import (
+	"unicode/utf8"
+
+	"github.com/blevesearch/bleve/analysis"
+)
+
+func buildTokenFromTerm(buffer []rune) *analysis.Token {
+	return &analysis.Token{
+		Term: buildTermFromRunes(buffer),
+	}
+}
+
+// TODO: Lifted from ngram_filter. Expose as public and re-use?
```

Comment:
This has been made public here:

https://github.com/blevesearch/bleve/blob/master/analysis/util.go#L37